### PR TITLE
HIPM-868: Support for Xamarin.Forms XAML Previewer

### PR DESCRIPTION
**Pull request by SvenEV**
[HIPM-868 on JIRA](https://atlassian-hip.cs.uni-paderborn.de/jira/browse/HIPM-868)

This PR adds classes to improve the design time experience when authoring XAML views. The classes are located in the namespace `PaderbornUniversity.SILab.Hip.Mobile.UI.DesignTime`.

For some pages this PR already includes sample data (i.e. sample view model instances) that is then displayed in the XAML previewer. For other pages, no sample data is available yet, but adding this data is straight forward and can be done on demand.

New types:
* Class `DesignMode`:
  * `DesignMode.ConfigureServices()` registers the mock services
  * `DesignMode.ConfigureViewModels()` sets up view models that contain sample data
  * Use `DesignMode.Initialize(this)` in the constructor of a page/view to set its binding context to such a sample view model (if available)
* Mock implementations for services required by some views (namespace `.DesignTime.Services`)
* Classes with sample data (namespace `.DesignTime.Data`)

### CODE REVIEW

File: Sources/HipMobileUI/DesignTime/DesignMode.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        /// This method should be called in the constructor of each page, right after the
+        /// "InitializeComponent()"-call.
+        /// </remarks>
+        /// <param name="view">View to set up for previewing</param>
+        public static void Initialize<TView>(TView view) where TView : BindableObject, IViewFor
+        {
+            if (view == null)
+                throw new ArgumentNullException(nameof(view));
+
+            var viewForType = view.GetType().GetTypeInfo().ImplementedInterfaces.FirstOrDefault(ii =>
```
Comment (cbruegg) [inline Sources/HipMobileUI/DesignTime/DesignMode.cs]: Should all of this work not also be skipped if `!IsEnabled`? Of course I haven't run any benchmarks, but especially using reflection might have an effect on the app's performance.

Comment (SvenEV) [reply Sources/HipMobileUI/DesignTime/DesignMode.cs]: Yep, I agree

**Comment by cbruegg** [comment]
@SvenEV Can this be merged now? Otherwise we'll need to update the description of PR #209 that this needs to be done before merging that.

**Comment by SvenEV** [comment]
@cbruegg I'd suggest merging #209 first because the TFS builds are currently failing here and #290 contains the necessary fixes/workarounds.

**Pull request merged**